src/components/product/utils/validationSchema.ts

```diff
@@ -241,68 +243,120 @@ export const commonCoreTermsOfSaleValidationSchema = {
-      value: Yup.string(),
-      label: Yup.string()
-    }).required(validationMessage.required),
-    minBalance: Yup.string().when(["tokenType", "tokenCriteria"], {
-      is: (tokenType: SelectDataProps, tokenCriteria: SelectDataProps) =>
-        tokenType?.value === TOKEN_TYPES[0].value ||
-        tokenType?.value === TOKEN_TYPES[2].value ||
-        (tokenType?.value === TOKEN_TYPES[1].value &&
-          tokenCriteria?.value === TOKEN_CRITERIA[0].value),
-      then: (schema) =>
-        schema
-          .required(validationMessage.required)
-          .matches(
-            /^\+?[1-9]\d*$/,
-            "Min balance must be greater than or equal to 1 (do not include commas/periods)"
-          )
-          .typeError("Value must be an integer greater than or equal to 1")
-    }),
-    tokenId: Yup.string().when(["tokenType", "tokenCriteria"], {
-      is: (tokenType: SelectDataProps, tokenCriteria: SelectDataProps) =>
-        tokenType?.value === TOKEN_TYPES[2].value ||
-        (tokenType?.value === TOKEN_TYPES[1].value &&
-          tokenCriteria?.value === TOKEN_CRITERIA[1].value),
-      then: (schema) =>
-        schema
-          .test(
-            "",
-            "Value must greater than or equal to 0 or a hex value up to 64 chars",
-            (value) => {
-              if (!value) {
-                return false;
+export type TokenGatingValidationSchema = ReturnType<
+  typeof getTokenGatingValidationSchema
+>;
+export const getTokenGatingValidationSchema = ({
+  coreSDK
+}: {
+  coreSDK: CoreSDK;
+}) =>
+  Yup.object({
+    tokenGating: Yup.object({
+      tokenContract: Yup.string()
+        .required(validationMessage.required)
+        .test("FORMAT", "Must be a valid address", (value) =>
+          value ? ethers.utils.isAddress(value) : true
+        )
+        .test("wrongTokenAddress", async function (tokenContract) {
+          const { tokenType } = this.parent;
+          if (
+            tokenType.value &&
+            tokenContract &&
+            ethers.utils.isAddress(tokenContract)
+          ) {
+            try {
+              switch (tokenType.value) {
+                case TokenTypes.erc20:
+                  {
+                    await coreSDK.erc20BalanceOf({
+                      contractAddress: tokenContract,
+                      owner: ethers.constants.AddressZero
+                    });
+                  }
+                  break;
+                case TokenTypes.erc721:
+                  {
+                    await coreSDK.erc721OwnerOf({
+                      contractAddress: tokenContract,
+                      tokenId: "1"
+                    });
+                  }
+                  break;
+                case TokenTypes.erc1155:
+                  {
+                    await coreSDK.erc1155BalanceOf({
+                      contractAddress: tokenContract,
+                      owner: ethers.constants.AddressZero,
+                      tokenId: "1"
```

Comment:
Potentially same problem than with ERC721 (depending on the contract implementation)